src/languages/ruby.js

```diff
@@ -11,13 +11,7 @@ export default function(hljs) {
   const regex = hljs.regex;
   const RUBY_METHOD_RE = '([a-zA-Z_]\\w*[!?=]?|[-+~]@|<<|>>|=~|===?|<=>|[<>]=?|\\*\\*|[-/+%^&*~`|]|\\[\\]=?)';
   // TODO: move concepts like CAMEL_CASE into `modes.js`
-  const CLASS_NAME_RE = regex.either(
-    /\b([A-Z]+[a-z0-9]+)+/,
-    // ends in caps
-    /\b([A-Z]+[a-z0-9]+)+[A-Z]+/,
-  )
-  ;
-  const CLASS_NAME_WITH_NAMESPACE_RE = regex.concat(CLASS_NAME_RE, /(::\w+)*/)
+  const CLASS_NAME_RE = /\b([A-Z]+[a-z0-9_]+)+[A-Z]*/;
```

Comment:
Doesn't this break (at least) the inheritance highlighting?

Reply:
Can you give an example syntax of what you mean?

```ruby
Class::With::Namespace
```
will work fine and better, because it doesn't highlight the `::` much like it does here in github, or vscode.

Reply:
```
class C::E < A::B
end
```

We have special rules to match syntax where we KNOW an identifier is a class (or class with namespaces).

>  it doesn't here in github, or vscode

Striving for exact matches against other highlighting tools isn't a thing we care about here.

---

In this case I'm sympathetic to not highlighting the `::`, but only if it can be done without adding a lot of complexity or advanced parsing to the grammar.

Reply:
You really want something like `[class](::[class])*` but our multi-match engine can't really handle that - with discrete coloring of the different pieces.  Maybe if you tried a long sequence with some items that matched 0 length, but I'm not sure I've tested multi-matching in that type of scenario before.

Reply:
This case would've worked fine, except it didn't recognise single-letter classes. I've adjusted it to do so.
```
class C::E < A::B
end
```

<img width="393" alt="image" src="https://github.com/user-attachments/assets/7a4907af-b78f-4104-9538-d11e404eb06e" />

Reply:
Please use "show the structure". as you develop.  If you're matching all those "one off" with just your class rule then you're missing out on the fact that the portion after the `<` is not just a `title.class`, it's a `title.class.inherited`.

This is why the simple multi-match rules exist - to flesh out more detail.

Reply:
I wasn't aware of the different handling for `title.class.inherited`. 

Whilst special casing `class A` works using a multi-match rule, it wouldn't help for `A.method(foo)` which will still be matched as a constant. So I'd suggest we simplify and not handle the `class A` case at all.

In the current version, it doesn't work either: https://highlightjs.org/demo#lang=ruby&v=1&theme=atom-one-dark&code=Y2xhc3MgQQplbmQKCscNOjpCywsgPCBDOjpExxJGb286OkZvbyA8yQvKGsYVyEjEC8VXQSA9ICdYJw%3D%3D

Reply:
<img width="1151" alt="image" src="https://github.com/user-attachments/assets/c8e541e8-186c-49a1-a584-cd49820b8abd" />